minorminer/utils/zephyr/node_edge.py

```diff
@@ -0,0 +1,823 @@
+    ) -> set[ZNode]:
+        """Returns the set of internal neighbors of self when restricted by ``where``.
+
+        Args:
+            where (Callable[[CartesianCoord | ZephyrCoord], bool]):
+                A coordinate filter. Applies to ``ccoord`` if :py:attr:`self.convert_to_z` is ``False``,
+                or to ``zcoord`` if :py:attr:`self.convert_to_z` is ``True``. Defaults to always.
+        Returns:
+            set[ZNode]: Set of internal neighbors of self when restricted by ``where``.
+        """
+        return set(self.neighbors_generator(nbr_kind=EdgeKind.INTERNAL, where=where))
+
+    def external_neighbors(
+        self,
+        where: Callable[[CartesianCoord | ZephyrCoord], bool] = lambda coord: True,
+    ) -> set[ZNode]:
+        """Returns the set of external neighbors of self when restricted by ``where``.
+
+        Args:
+            where (Callable[[CartesianCoord | ZephyrCoord], bool]):
+                A coordinate filter. Applies to ``ccoord`` if :py:attr:`self.convert_to_z` is ``False``,
+                or to ``zcoord`` if :py:attr:`self.convert_to_z` is ``True``. Defaults to always.
+        Returns:
+            set[ZNode]: Set of external neighbors of self when restricted by ``where``.
+        """
+        return set(self.neighbors_generator(nbr_kind=EdgeKind.EXTERNAL, where=where))
+
+    def odd_neighbors(
+        self,
+        where: Callable[[CartesianCoord | ZephyrCoord], bool] = lambda coord: True,
+    ) -> set[ZNode]:
+        """Returns the set of odd neighbors of self when restricted by ``where``.
+
+        Args:
+            where (Callable[[CartesianCoord | ZephyrCoord], bool]):
+                A coordinate filter. Applies to ``ccoord`` if :py:attr:`self.convert_to_z` is ``False``,
+                or to ``zcoord`` if :py:attr:`self.convert_to_z` is ``True``. Defaults to always.
+        Returns:
+            set[ZNode]: Set of odd neighbors of self when restricted by ``where``.
+        """
+        return set(self.neighbors_generator(nbr_kind=EdgeKind.ODD, where=where))
+
+    def is_internal_neighbor(
+        self,
+        other: ZNode,
+        where: Callable[[CartesianCoord | ZephyrCoord], bool] = lambda coord: True,
+    ) -> bool:
+        """Returns whether other is an internal neighbor of self when restricted by ``where``.
+
+        Args:
+            other (ZNode): Another instance to compare with self.
+            where (Callable[[CartesianCoord | ZephyrCoord], bool]):
+                A coordinate filter. Applies to ``ccoord`` if :py:attr:`self.convert_to_z` is ``False``,
+                or to ``zcoord`` if :py:attr:`self.convert_to_z` is ``True``. Defaults to always.
+
+        Returns:
+            bool: Whether other is an internal neighbor of self when restricted by ``where``.
+        """
+        for nbr in self.internal_neighbors_generator(where=where):
+            if other == nbr:
+                return True
+        return False
```

Comment:
This, and the similar 3 other functions feel redundant as they can be replaced with:

```suggestion
        return other in self.internal_neighbors_generator(where=where)
```